### PR TITLE
rpm: support deletion of stale metadata

### PR DESCRIPTION
**Pull request by urbanchef**
The repo metadata needs to be updated when RPM packages are deleted from
s3 backend for a some reason. Now `mkrepo` supports not only adding, but
also deleting the metadata and does it in the same iteration. To make it
possible, `mkrepo` deletes the metadata first, then adds it.

Part of #50

### CODE REVIEW

**Comment by Totktonada** [comment]
Seems related to #50.

File: rpmrepo.py [hunk 1]
```diff
@@ -1293,10 +1293,25 @@ def update_repo(storage, sign, tempdir, force=False):
         existing_files.add((file_path, mtime))
 
     files_to_add = existing_files - recorded_files
+    stale_rpms_to_delete = recorded_files - existing_files
     # List of packages that can't be added to the index
     # (some problems encountered during processing).
     malformed_list = []
 
+    if stale_rpms_to_delete:
```
Comment (ylobankov) [inline rpmrepo.py:1301]: I think we can optimize it to something like this:

```python
if files_to_delete:
    stale_primary = {}
    copied_primary = copy.deepcopy(primary)
    for file_to_delete in files_to_delete:
        for primary_nerv, primary_value in list(copied_primary.items())[:]:
            if primary_value['location'] == file_to_delete[0]:
                stale_primary[primary_nerv] = primary_value
                del copied_primary[primary_nerv]
                break

    ...
```

Basically, we delete items that have been already added to `stale_primary`.

**Review by ylobankov — COMMENTED** [review]
Please add to the commit message `Part of #50`.

File: rpmrepo.py (hunk 1 above)
Comment (urbanchef) [reply rpmrepo.py]: Done

**Comment by urbanchef** [comment]
> Please add to the commit message `Part of #50`.

Done

**Pull request merged**